src/integration-test/java/org/openlmis/stockmanagement/service/CalculatedStockOnHandServiceIntegrationTest.java

```diff
@@ -42,38 +47,123 @@
   @Autowired
   private CalculatedStockOnHandService calculatedStockOnHandService;
 
+  @Autowired
+  private CalculatedStockOnHandRepository calculatedStockOnHandRepository;
+
+  @Autowired
+  private StockCardRepository stockCardRepository;
+
+  @Autowired
+  private StockEventsRepository stockEventsRepository;
+
+  private UUID facility = randomUUID();
+  private UUID program = randomUUID();
+  private UUID product = randomUUID();
+  private UUID lot = randomUUID();
+
+  private int quantity = new Random().nextInt();
+  private int quantity2 = new Random().nextInt();
+
+  private StockCard stockCard;
+
+  @Before
+  public void setUp() {
+    StockEvent event = new StockEventDataBuilder()
+        .withoutId()
+        .withFacility(facility)
+        .withProgram(program)
+        .build();
+    stockEventsRepository.save(event);
+
+    stockCard = new StockCardDataBuilder(event)
+        .withoutId()
+        .withOrderable(product)
+        .withLot(lot)
+        .build();
+    stockCardRepository.save(stockCard);
+  }
+
   @Test
-  public void shouldSaveCalculatedStockOnHandFromStockCard() {
-    //given
-    int quantity = 20;
-    StockCard stockCard = getStockCard(randomUUID(), randomUUID(), randomUUID(), randomUUID(),
-            quantity);
+  public void shouldGetLatestStockCardsWithStockOnHandWhenDateIsNotSet() {
+    CalculatedStockOnHand calculatedStockOnHand = new CalculatedStockOnHand(
+        quantity, stockCard, LocalDate.now(), ZonedDateTime.now());
 
-    //when
-    CalculatedStockOnHand result = calculatedStockOnHandService.saveFromStockCard(stockCard);
+    calculatedStockOnHandRepository.save(calculatedStockOnHand);
 
-    //then
-    assertThat(result.getStockOnHand(), is(quantity));
+    List<StockCard> foundStockCards = calculatedStockOnHandService
+        .getStockCardsWithStockOnHand(program, facility);
+
+    assertThat(foundStockCards.size(), is(1));
+    assertThat(foundStockCards.get(0).getStockOnHand(), is(quantity));
+    assertThat(foundStockCards.get(0).getOccurredDate(), is(LocalDate.now()));
```

Comment:
This is a local date so it does not contain time value, the only way it would fail is when this test would start in one day and finish in another and there is a lot of tests in stockmanagement or other repositories would fail. Nevertheless, I can change this.